src/transports/ssh.c

```diff
@@ -893,9 +913,24 @@ int git_transport_ssh_with_paths(git_transport **out, git_remote *owner, void *p
 #endif
 }
 
+void git_transport_ssh_global_shutdown(void);
+
 int git_transport_ssh_global_init(void)
 {
 #ifdef GIT_SSH
+#ifdef GIT_WIN32
+	const char *(*libssh2_version)(int) = NULL;
+#else
+	libssh2_handle = dlopen("libssh2.dylib", RTLD_LAZY);
+	const char *(*libssh2_version)(int) = dlsym(libssh2_handle, "libssh2_version");
```

Comment:
Can you get away without a `libssh2_version` function pointer if you use `RTLD_NOW | RTLD_GLOBAL` in the dlopen? (as outlined in the end of the answer http://stackoverflow.com/questions/20658809/dynamic-loading-and-weak-symbol-resolution )